crates/ruff_linter/src/rules/airflow/rules/removal_in_3.rs

```diff
@@ -849,3 +954,55 @@ fn is_airflow_builtin_or_provider(segments: &[&str], module: &str, symbol_suffix
         _ => false,
     }
 }
+
+fn is_task_context_referenced(checker: &mut Checker, expr: &Expr) -> bool {
+    let parents: Vec<_> = checker.semantic().current_statements().collect();
+
+    for stmt in parents {
+        if let Stmt::FunctionDef(function_def) = stmt {
+            if is_decorated_with(checker, function_def) {
+                let arguments = extract_task_function_arguments(function_def);
+
+                for deprecated_arg in REMOVED_CONTEXT_KEYS {
+                    if arguments.contains(&deprecated_arg.to_string()) {
+                        checker.diagnostics.push(Diagnostic::new(
+                            Airflow3Removal {
+                                deprecated: deprecated_arg.to_string(),
+                                replacement: Replacement::None,
+                            },
+                            expr.range(),
+                        ));
+                        return true;
+                    }
+                }
+            }
+        }
+    }
+
+    false
+}
+
+fn extract_task_function_arguments(stmt: &StmtFunctionDef) -> Vec<String> {
+    let mut arguments = Vec::new();
+
+    for param in &stmt.parameters.args {
+        arguments.push(param.parameter.name.to_string());
+    }
+
+    if let Some(vararg) = &stmt.parameters.kwarg {
+        arguments.push(format!("**{}", vararg.name));
+    }
+
+    arguments
+}
+
+fn is_decorated_with(checker: &mut Checker, stmt: &StmtFunctionDef) -> bool {
+    stmt.decorator_list.iter().any(|decorator| {
+        checker
+            .semantic()
+            .resolve_qualified_name(map_callable(&decorator.expression))
+            .is_some_and(|qualified_name| {
+                matches!(qualified_name.segments(), ["airflow", "decorators", "task"])
+            })
+    })
+}
```

Comment:
Based on the name of the function, did you mean to make this a generic function over any decorator originating from `airflow.decorators.*` ? If not, can you rename this to `has_task_decorator` ?

Reply:
Yes, the intent was to check only for the @task decorator. I have renamed this function.